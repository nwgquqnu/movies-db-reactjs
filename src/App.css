.App {
  text-align: center;
  background: #888888;
  min-height: 100vh;
  font-family: Montserrat, Arial, Helvetica, sans-serif;
  font-style: normal;
  color: #FFFFFF;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background: url("./Bitmap.png"), linear-gradient(0deg, #000000, #000000);
  background-position: center;
  min-height: 40vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

.App-header, .App-main {
  max-width: 1200px;
  margin: 0 auto
}

.App-main {
  margin-top: 10px;
  background: #232323;
  min-height: calc(100vh - 40vh - 10px);
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
