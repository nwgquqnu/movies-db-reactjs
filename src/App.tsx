import * as React from 'react';
import * as css from './App.module.css';
import Footer from './components/Footer';
import Header from './components/Header';
import Main from './components/Main';

function App() {
  return (
    <div className={css.app}>
      <Header />
      <Main />
      <Footer />
    </div>
  );
}

export default App;
