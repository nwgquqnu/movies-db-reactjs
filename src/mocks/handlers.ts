import { rest } from 'msw'

const baseFetchUrl = "http://localhost:4000"
const movie = {
    id: 123,
    title: "movie title",
    tagline: "movie tagline",
    vote_average: 0,
    vote_count: 1,
    release_date: "movie release_date",
    poster_path: "movie poster_path",
    overview: "movie overview",
    budget: 2,
    revenue: 3,
    runtime: 4,
    genres: ["Action"],
};

export const handlers = [
    rest.get(`${baseFetchUrl}/movies`, (req, res, ctx) => res(
        ctx.status(200),
        ctx.json({
            data: [movie]
        })
    )),
    rest.get(`${baseFetchUrl}/movies/123`, (req, res, ctx) => res(
        ctx.status(200),
        ctx.json(movie)
    )),
];