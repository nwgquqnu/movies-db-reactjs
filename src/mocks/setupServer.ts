import { server } from './server';
import { default as crossFetch } from 'cross-fetch';

// import('node-fetch').then(module => globalThis.fetch = module.default as any);
globalThis.fetch = crossFetch;
// Establish API mocking before all tests.
beforeAll(() => server.listen())

// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => server.resetHandlers())

// Clean up after the tests are finished.
afterAll(() => server.close())