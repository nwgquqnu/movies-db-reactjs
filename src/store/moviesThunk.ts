import { createAsyncThunk, PayloadAction } from '@reduxjs/toolkit'
import { MovieApi } from '../Services/MovieDataFetcher';
import { ActionType, FetchedGenresAction, FetchedMoviesAction } from '../types/movieActions';
import { FetchApiMoviesParams, Movie, MovieDbState, NewMovie, newMovieFields, SortOrder } from '../types/movieModels';
import { AppDispatch } from './store';

type FetchMoviesParams = {
    sortOrder?: SortOrder;
    search?: string;
    activeGenre?: string | null;
}

type ActionPayload = {
    list: ReadonlyArray<Movie>,
    transformations: FetchApiMoviesParams,
};

//synchronous action creator
const fetchMoviesSuccess = (movies: ReadonlyArray<Movie>, params: FetchApiMoviesParams): FetchedMoviesAction => ({
    type: ActionType.FetchedMovies,
    payload: { list: movies, params }
});

/*asynchronous thunk action creator
  calls the api, then dispatches the synchronous action creator
*/
export const fetchMovies = (fetchParams: FetchMoviesParams) => {
    return async (dispatch: AppDispatch, getState: () => MovieDbState) => {
        try {
            const state = getState();
            const sortOrder = fetchParams.sortOrder !== undefined ? fetchParams.sortOrder : state.sortOrder;
            const activeGenre = fetchParams.activeGenre !== undefined ? fetchParams.activeGenre : state.activeGenre;
            const params: FetchApiMoviesParams = {
                search: fetchParams.search !== undefined ? fetchParams.search : state.searchText,
                searchBy: fetchParams.search !== undefined ? "title" : state.searchBy,
                sortBy: getSortBy(sortOrder),
                sortOrder: getSortDirection(sortOrder),
                filter: activeGenre ? [activeGenre] : undefined,
            };
            let movies = await MovieApi.fetchMovieData(
                params.sortBy,
                params.sortOrder,
                params.search,
                params.searchBy,
                params.filter
            );
            dispatch(fetchMoviesSuccess(movies, params));
        } catch (e) {
            console.log(e)
        }
    }
};

/*asynchronous thunk action creator
  calls the api, then dispatches the synchronous action creator
*/
export const addMovie = (movie: NewMovie) => {
    return async (dispatch: AppDispatch) => {
        try {
            await MovieApi.addMovie(movie);
            dispatch(fetchMovies({}));
        } catch (e) {
            console.log(e)
        }
    }
};
/*asynchronous thunk action creator
  calls the api, then dispatches the synchronous action creator
*/
export const updateMovie = (movie: Movie) => {
    return async (dispatch: AppDispatch) => {
        try {
            await MovieApi.updateMovie(movie);
            dispatch(fetchMovies({}));
        } catch (e) {
            console.log(e)
        }
    }
};
/*asynchronous thunk action creator
  calls the api, then dispatches the synchronous action creator
*/
export const deleteMovie = (movie: Movie) => {
    return async (dispatch: AppDispatch) => {
        try {
            await MovieApi.deleteMovie(movie);
            dispatch(fetchMovies({}));
        } catch (e) {
            console.log(e)
        }
    }
};

//synchronous action creator
const fetchGenresSuccess = (genres: ReadonlyArray<string>): FetchedGenresAction => ({
    type: ActionType.FetchedGenres,
    payload: genres
});

/*asynchronous thunk action creator
  calls the api, then dispatches the synchronous action creator
*/
export const fetchGenres = () => {
    return async (dispatch: AppDispatch) => {
        try {
            let genres = await MovieApi.fetchGenreList();
            dispatch(fetchGenresSuccess(genres))
        } catch (e) {
            console.log(e)
        }
    }
};

// export const fetchMoviesAT = createAsyncThunk<ActionPayload, FetchApiMoviesParams, { state: MovieDbState }>(
//     ActionType.FetchedMovies,
//     async (config: FetchApiMoviesParams, { getState }) => {
//         const state = getState();
//         const response = await MovieApi.fetchMovieData(
//             getSortBy(state.sortOrder) || config.sortBy,
//             getSortDirection(state.sortOrder) || config.sortOrder,
//             config.search,
//             config.searchBy,
//             state.activeGenre && [state.activeGenre] || config.filter
//         );
//         // The value we return becomes the `fulfilled` action payload
//         return {
//             list: response,
//             transformations: config,
//         };
//     }
// );

function getSortBy(order: SortOrder): string {
    switch (order) {
        case SortOrder.ByNameAsc:
        case SortOrder.ByNameDesc:
            return newMovieFields.title;
        case SortOrder.ByReleaseAsc:
        case SortOrder.ByReleaseDesc:
            return newMovieFields.release_date;
        default:
            const _exhaustiveCheck: never = order;
            return _exhaustiveCheck;
    }
}

function getSortDirection(order: SortOrder): "asc" | "desc" {
    switch (order) {
        case SortOrder.ByNameAsc:
        case SortOrder.ByReleaseAsc:
            return "asc";
        case SortOrder.ByNameDesc:
        case SortOrder.ByReleaseDesc:
            return "desc";
        default:
            const _exhaustiveCheck: never = order;
            return _exhaustiveCheck;
    }
}
