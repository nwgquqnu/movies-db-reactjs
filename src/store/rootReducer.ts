import { ActionType, MovieDbStateAction } from "../types/movieActions";
import { FetchApiMoviesParams, FetchedMoviesPayload, Movie, MovieDbState, SortOrder, UpdateActivity } from "../types/movieModels";

const initialState: Readonly<MovieDbState> = {
    moviesList: [],
    genres: [],
    sortOrder: SortOrder.ByNameAsc
}

export function rootReducer(state: Readonly<MovieDbState> = initialState, action: Readonly<MovieDbStateAction>): Readonly<MovieDbState> {
    switch (action.type) {
        case ActionType.SelectMovie:
            return handleSelectMovie(state, action.payload);
        case ActionType.ShowAddMovie:
            return handleShowHideAction(state, UpdateActivity.addActivity);
        case ActionType.ShowEditMovie:
            return handleShowHideAction(state, UpdateActivity.editActivity, action.payload);
        case ActionType.ShowDeleteMovie:
            return handleShowHideAction(state, UpdateActivity.deleteActivity, action.payload);
        case ActionType.HideMovieUpdate:
            return handleShowHideAction(state);
        case ActionType.FetchedMovies:
            return handleFetchedMovies(state, action.payload);
        case ActionType.FetchedGenres:
            return handleFetchedGenres(state, action.payload);
        default:
            const _exhaustiveCheck: never = action;
            return state;
    }
}

function handleFetchedGenres(state: Readonly<MovieDbState>, payload: ReadonlyArray<string>): Readonly<MovieDbState> {
    return {
        ...state,
        genres: payload,
    }
}

function handleFetchedMovies(state: Readonly<MovieDbState>, payload: FetchedMoviesPayload): Readonly<MovieDbState> {
    const movies = payload.list;
    const params = payload.params;
    const selectedMovie = getMovie(state.selectedMovie, movies);
    return {
        ...state,
        moviesList: movies,
        sortOrder: getSortOrder(params),
        activeGenre: params.filter?.shift(),
        selectedMovie,
        searchText: params.search,
        searchBy: params.searchBy,
    }
}

function getMovie(target: Movie | undefined, list: ReadonlyArray<Movie>): Movie | undefined {
    if (!target) {
        return;
    }
    return list.find(movie => movie.id === target.id);
}

function getSortOrder(params: FetchApiMoviesParams): SortOrder {
    if (params.sortOrder === "desc") {
        if (params.sortBy === "title") {
        return SortOrder.ByNameDesc
        } else {
            return SortOrder.ByReleaseDesc;
        }
    } else if (params.sortBy === "release") {
        return SortOrder.ByReleaseAsc;
    }
    return SortOrder.ByNameAsc;
}


function handleSelectMovie(state: Readonly<MovieDbState>, movie: Readonly<Movie> | undefined): Readonly<MovieDbState> {
    const selectedMovie = movie;
    return { ...state, selectedMovie };
}

function handleShowHideAction(state: Readonly<MovieDbState>,
    currentUpdateActivity?: UpdateActivity,
    movieUnderUpdateActivity?: Readonly<Movie>): Readonly<MovieDbState> {
    return { ...state, currentUpdateActivity, movieUnderUpdateActivity };
}
