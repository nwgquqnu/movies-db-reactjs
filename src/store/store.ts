import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterReducer from './counterSlice'
import moviesSlice from './moviesSlice';
import { rootReducer } from './rootReducer';

const initialState = { value: 0 }


export const store = configureStore({
    reducer: rootReducer,
    // preloadedState: initialState,
    

    // reducer: {
    //   counter: counterReducer,
    //   movies: moviesSlice,
    // }
  });

  // Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;