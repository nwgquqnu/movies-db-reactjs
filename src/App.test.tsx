import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import App from "./App";
import "./mocks/setupServer";
import { store } from "./store/store";

window.scrollTo = jest.fn();

describe("Application", () => {
    it("snapshot is matched on root path", async () => {
        const { asFragment } = render(
            <Provider store={store}>
                <App />
            </Provider>, { wrapper: MemoryRouter });
        expect(await screen.findByText(/movie title/)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
    it("snapshot is matched on movie details path", async () => {
        const { asFragment } = render(
            <Provider store={store}>
                <App />
            </Provider>, { wrapper: MemoryRouter });
        const title = await screen.findByText(/movie title/);
        expect(title).toBeInTheDocument();
        userEvent.click(title);

        expect(await screen.findByText(/movie overview/)).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();

    });

});