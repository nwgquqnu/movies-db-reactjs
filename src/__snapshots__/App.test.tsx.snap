// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application snapshot is matched on movie details path 1`] = `
<DocumentFragment>
  <div>
    <header>
      <strong>
        <span>
          netflix
        </span>
        roulette
      </strong>
      <button>
        X
      </button>
      <img
        alt="movie title"
        src="movie poster_path"
      />
      <div>
        <h1>
          movie title
          <span>
            0.0
          </span>
        </h1>
        <p>
          Action
        </p>
        <p>
          <span>
            movi
          </span>
          <span>
            0h 04m
          </span>
        </p>
        <p>
          movie overview
        </p>
      </div>
    </header>
    <main>
      <nav>
        <div>
          <input
            checked=""
            class="radio-toggle"
            id="radio-toggle-All"
            name="genres"
            type="radio"
            value="All"
          />
          <label
            for="radio-toggle-All"
          >
            All
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Action"
            name="genres"
            type="radio"
            value="Action"
          />
          <label
            for="radio-toggle-Action"
          >
            Action
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Adventure"
            name="genres"
            type="radio"
            value="Adventure"
          />
          <label
            for="radio-toggle-Adventure"
          >
            Adventure
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Animation"
            name="genres"
            type="radio"
            value="Animation"
          />
          <label
            for="radio-toggle-Animation"
          >
            Animation
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Comedy"
            name="genres"
            type="radio"
            value="Comedy"
          />
          <label
            for="radio-toggle-Comedy"
          >
            Comedy
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Crime"
            name="genres"
            type="radio"
            value="Crime"
          />
          <label
            for="radio-toggle-Crime"
          >
            Crime
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Documentary"
            name="genres"
            type="radio"
            value="Documentary"
          />
          <label
            for="radio-toggle-Documentary"
          >
            Documentary
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Drama"
            name="genres"
            type="radio"
            value="Drama"
          />
          <label
            for="radio-toggle-Drama"
          >
            Drama
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Family"
            name="genres"
            type="radio"
            value="Family"
          />
          <label
            for="radio-toggle-Family"
          >
            Family
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Fantasy"
            name="genres"
            type="radio"
            value="Fantasy"
          />
          <label
            for="radio-toggle-Fantasy"
          >
            Fantasy
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-History"
            name="genres"
            type="radio"
            value="History"
          />
          <label
            for="radio-toggle-History"
          >
            History
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Horror"
            name="genres"
            type="radio"
            value="Horror"
          />
          <label
            for="radio-toggle-Horror"
          >
            Horror
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Music"
            name="genres"
            type="radio"
            value="Music"
          />
          <label
            for="radio-toggle-Music"
          >
            Music
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Mystery"
            name="genres"
            type="radio"
            value="Mystery"
          />
          <label
            for="radio-toggle-Mystery"
          >
            Mystery
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Romance"
            name="genres"
            type="radio"
            value="Romance"
          />
          <label
            for="radio-toggle-Romance"
          >
            Romance
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Science-Fiction"
            name="genres"
            type="radio"
            value="Science Fiction"
          />
          <label
            for="radio-toggle-Science-Fiction"
          >
            Science Fiction
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-TV-Movie"
            name="genres"
            type="radio"
            value="TV Movie"
          />
          <label
            for="radio-toggle-TV-Movie"
          >
            TV Movie
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Thriller"
            name="genres"
            type="radio"
            value="Thriller"
          />
          <label
            for="radio-toggle-Thriller"
          >
            Thriller
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-War"
            name="genres"
            type="radio"
            value="War"
          />
          <label
            for="radio-toggle-War"
          >
            War
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Western"
            name="genres"
            type="radio"
            value="Western"
          />
          <label
            for="radio-toggle-Western"
          >
            Western
          </label>
        </div>
        <label>
          <span>
            sort by
          </span>
          <select>
            <option
              value="BY_RELEASE_ASC"
            >
              Release Date
            </option>
            <option
              value="BY_NAME_ASC"
            >
              Name
            </option>
            <option
              value="BY_RELEASE_DESC"
            >
              Release Date Descending
            </option>
            <option
              value="BY_NAME_DESC"
            >
              Name Descending
            </option>
          </select>
        </label>
      </nav>
      <div>
        <strong>
          1
        </strong>
         movie found
      </div>
      <div>
        <article>
          <nav
            class=""
            role="navigation"
          >
            <ul>
              <li>
                <button
                  aria-haspopup="true"
                >
                  Details
                </button>
                <ul
                  aria-label="submenu"
                  class="dropdown"
                >
                  <li>
                    <button>
                      Edit
                    </button>
                  </li>
                  <li>
                    <button>
                      Delete
                    </button>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
          <img
            alt="movie title"
            src="movie poster_path"
          />
          <div>
            <h1>
              movie title
            </h1>
            <p>
              movi
            </p>
            <p>
              Action
            </p>
          </div>
        </article>
      </div>
    </main>
    <footer>
      <strong>
        <span>
          netflix
        </span>
        roulette
      </strong>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Application snapshot is matched on root path 1`] = `
<DocumentFragment>
  <div>
    <header>
      <strong>
        <span>
          netflix
        </span>
        roulette
      </strong>
      <button>
        + Add Movie
      </button>
      <form>
        <label>
          <span>
            Find your movie
          </span>
          <input
            placeholder="What do you want to watch?"
            type="text"
            value=""
          />
        </label>
        <button
          class=""
        >
          Search
        </button>
      </form>
    </header>
    <main>
      <nav>
        <div>
          <input
            checked=""
            class="radio-toggle"
            id="radio-toggle-All"
            name="genres"
            type="radio"
            value="All"
          />
          <label
            for="radio-toggle-All"
          >
            All
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Action"
            name="genres"
            type="radio"
            value="Action"
          />
          <label
            for="radio-toggle-Action"
          >
            Action
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Adventure"
            name="genres"
            type="radio"
            value="Adventure"
          />
          <label
            for="radio-toggle-Adventure"
          >
            Adventure
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Animation"
            name="genres"
            type="radio"
            value="Animation"
          />
          <label
            for="radio-toggle-Animation"
          >
            Animation
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Comedy"
            name="genres"
            type="radio"
            value="Comedy"
          />
          <label
            for="radio-toggle-Comedy"
          >
            Comedy
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Crime"
            name="genres"
            type="radio"
            value="Crime"
          />
          <label
            for="radio-toggle-Crime"
          >
            Crime
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Documentary"
            name="genres"
            type="radio"
            value="Documentary"
          />
          <label
            for="radio-toggle-Documentary"
          >
            Documentary
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Drama"
            name="genres"
            type="radio"
            value="Drama"
          />
          <label
            for="radio-toggle-Drama"
          >
            Drama
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Family"
            name="genres"
            type="radio"
            value="Family"
          />
          <label
            for="radio-toggle-Family"
          >
            Family
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Fantasy"
            name="genres"
            type="radio"
            value="Fantasy"
          />
          <label
            for="radio-toggle-Fantasy"
          >
            Fantasy
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-History"
            name="genres"
            type="radio"
            value="History"
          />
          <label
            for="radio-toggle-History"
          >
            History
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Horror"
            name="genres"
            type="radio"
            value="Horror"
          />
          <label
            for="radio-toggle-Horror"
          >
            Horror
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Music"
            name="genres"
            type="radio"
            value="Music"
          />
          <label
            for="radio-toggle-Music"
          >
            Music
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Mystery"
            name="genres"
            type="radio"
            value="Mystery"
          />
          <label
            for="radio-toggle-Mystery"
          >
            Mystery
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Romance"
            name="genres"
            type="radio"
            value="Romance"
          />
          <label
            for="radio-toggle-Romance"
          >
            Romance
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Science-Fiction"
            name="genres"
            type="radio"
            value="Science Fiction"
          />
          <label
            for="radio-toggle-Science-Fiction"
          >
            Science Fiction
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-TV-Movie"
            name="genres"
            type="radio"
            value="TV Movie"
          />
          <label
            for="radio-toggle-TV-Movie"
          >
            TV Movie
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Thriller"
            name="genres"
            type="radio"
            value="Thriller"
          />
          <label
            for="radio-toggle-Thriller"
          >
            Thriller
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-War"
            name="genres"
            type="radio"
            value="War"
          />
          <label
            for="radio-toggle-War"
          >
            War
          </label>
          <input
            class="radio-toggle"
            id="radio-toggle-Western"
            name="genres"
            type="radio"
            value="Western"
          />
          <label
            for="radio-toggle-Western"
          >
            Western
          </label>
        </div>
        <label>
          <span>
            sort by
          </span>
          <select>
            <option
              value="BY_RELEASE_ASC"
            >
              Release Date
            </option>
            <option
              value="BY_NAME_ASC"
            >
              Name
            </option>
            <option
              value="BY_RELEASE_DESC"
            >
              Release Date Descending
            </option>
            <option
              value="BY_NAME_DESC"
            >
              Name Descending
            </option>
          </select>
        </label>
      </nav>
      <div>
        <strong>
          1
        </strong>
         movie found
      </div>
      <div>
        <article>
          <nav
            class=""
            role="navigation"
          >
            <ul>
              <li>
                <button
                  aria-haspopup="true"
                >
                  Details
                </button>
                <ul
                  aria-label="submenu"
                  class="dropdown"
                >
                  <li>
                    <button>
                      Edit
                    </button>
                  </li>
                  <li>
                    <button>
                      Delete
                    </button>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
          <img
            alt="movie title"
            src="movie poster_path"
          />
          <div>
            <h1>
              movie title
            </h1>
            <p>
              movi
            </p>
            <p>
              Action
            </p>
          </div>
        </article>
      </div>
    </main>
    <footer>
      <strong>
        <span>
          netflix
        </span>
        roulette
      </strong>
    </footer>
  </div>
</DocumentFragment>
`;
